src/Integration/Exclusions/ExclusionSettingsFileStorage.cs

```
@@ -17,23 +17,84 @@
+using SonarLint.VisualStudio.Integration.NewConnectedMode;
+using SonarLint.VisualStudio.Integration.Resources;
 using SonarQube.Client.Models;
 
 namespace SonarLint.VisualStudio.Integration.Exclusions
 {
-    [Export(typeof(IExclusionSettingsFileStorage))]
     internal class ExclusionSettingsFileStorage : IExclusionSettingsFileStorage
     {
-        public void SaveSettings(ServerExclusions settings)
+        private readonly BindingConfiguration bindingConfiguration;
+        private readonly ILogger logger;
+        private readonly IFileSystem fileSystem;
+
+        public ExclusionSettingsFileStorage(ILogger logger, IFileSystem fileSystem, BindingConfiguration bindingConfiguration)
         {
+            this.logger = logger;
+            this.fileSystem = fileSystem;         
+            this.bindingConfiguration = bindingConfiguration;            
         }
 
         public ServerExclusions GetSettings()
         {
+            try
+            {
+                if(bindingConfiguration.Mode == SonarLintMode.Standalone)
+                {
+                    logger.WriteLine(Strings.ExclusionOnStandaloneNotSupported);
+                    return null;
+                }
+
+                if (!fileSystem.File.Exists(FilePath))
+                {
+                    logger.WriteLine(String.Format(Strings.ExclusionGetError, Strings.ExclusionFileNotFound));
+                    return null;
+                }
+
+                var fileContent = fileSystem.File.ReadAllText(FilePath);
+                return JsonConvert.DeserializeObject<ServerExclusions>(fileContent);
+            }
+            catch(Exception ex)
+            {
+                logger.WriteLine(String.Format(Strings.ExclusionGetError, ex.Message));
+            }
+
             return null;
         }
+
+        public void SaveSettings(ServerExclusions settings)
+        {
+            try
+            {
+                if (bindingConfiguration.Mode == SonarLintMode.Standalone)
```

Is this being part of the connected mode binding process? If so, how would this be standalone when we reach this point?

---

I think coding defensively is the wrong approach here; it means there is something wrong elsewhere in our logic.
An assertion would be better.

---

I understand but in this case, we would have the possibility to write a file to disk unnecessarily. Do we really want that?

---

Yes. We should not be coding round/hiding errors in our code, particularly new code. We should be making the failures obvious so we find them during development/dogfooding and fix them.